docs/contributing.rst

```diff
@@ -0,0 +1,200 @@
+   Python interpreter, and installation information. The version of Zarr can be obtained
+   from the ``zarr.__version__`` property. Please also state how Zarr was installed,
+   e.g., "installed via pip into a virtual environment", or "installed using conda".
+   Information about other packages installed can be obtained by executing ``pip list``
+   (if using pip to install packages) or ``conda list`` (if using conda to install
+   packages) from the operating system command prompt. The version of the Python
+   interpreter can be obtained by running a Python interactive session, e.g.::
+
+    $ python
+    Python 3.6.1 (default, Mar 22 2017, 06:17:05)
+    [GCC 6.3.0 20170321] on linux
+
+3. An explanation of why the current behaviour is wrong/not desired, and what you
+   expect instead.
+
+Enhancement proposals
+---------------------
+
+If you have an idea about a new feature or some other improvement to Zarr, please raise a
+`GitHub issue <https://github.com/alimanfoo/zarr/issues/new>`_ first to discuss.
+
+We very much welcome ideas and suggestions for how to improve Zarr, but please bear in
+mind that we are likely to be conservative in accepting proposals for new features. The
+reasons for this are that we would like to keep the Zarr code base lean and focused on
+a core set of functionalities, and available time for development, review and maintenance
+of new features is limited. But if you have a great idea, please don't let that stop
+you posting it on GitHub, just please don't be offended if we respond cautiously.
+
+Contributing code and/or documentation
+--------------------------------------
+
+Forking the repository
+~~~~~~~~~~~~~~~~~~~~~~
+
+The Zarr source code is hosted on GitHub at the following location:
+
+* `https://github.com/zarr-developers/zarr <https://github.com/zarr-developers/zarr>`_
+
+You will need your own fork to work on the code. Go to the link above and hit
+the "Fork" button. Then clone your fork to your local machine::
+
+    $ git clone git@github.com:your-user-name/zarr.git
+    $ cd zarr
+    $ git remote add upstream git@github.com:zarr-developers/zarr.git
+
+Creating a development environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To work with the Zarr source code, it is recommended to set up a Python virtual
+environment and install all Zarr dependencies using the same versions as are used by
+the core developers and continuous integration services. Assuming you have a Python
+3 interpreter already installed, and have also installed the virtualenv package, and
+you have cloned the Zarr source code and your current working directory is the root of
+the repository, you can do something like the following::
+
+    $ mkdir -p ~/pyenv/zarr-dev
+    $ virtualenv --no-site-packages --python=/usr/bin/python3.6 ~/pyenv/zarr-dev
+    $ source ~/pyenv/zarr-dev/bin/activate
+    $ pip install -r requirements_dev.txt
+    $ pip install -r requirements_dev_optional.txt
+    $ pip install -e .
+
```

Comment:
I would suggest also including an example of how to do this with conda. My experience is that this is a common workflow.

```
$ conda env create -n zarr-dev -f requirements_dev.yml  # req file would need to be added
$ source activate zarr-dev
$ pip install -e  # or python setup.py develop
```

Reply:
I didn't give a conda example as (last time I looked) you can't use tox with conda, and tox is the best way at the moment to run all the tests (including doctests, coverage, flake8 etc.) and verify support for PY2 and PY3.

Reply:
`virtualenv` has been packaged and exists on `defaults` and `conda-forge`. Once that was done `tox` was added to `conda-forge` in short order.

Reply:
I think I'll leave this as-is for now as I don't have time to verify all the steps to get tox fully up and running all test environments under a conda installation, but if someone wants to do this later and can provide the necessary commands then very happy to add.

Reply:
Sure. That was only meant as an informative comment. Happy with this as is.